### PR TITLE
Barrels can now be destroyed.

### PR DESCRIPTION
**Pull request by Irradiation**
This incidentally also prevents you from stuffing a gazillion (hostile) mobs into barrels without them being able to escape. I have paid my dues to the Kurfurst metaclub.

[content]

:cl:
 * rscadd: You (and mobs) can now break barrels.

### CODE REVIEW

File: code/modules/reagents/reagent_dispenser.dm [hunk 1]
```diff
@@ -449,6 +460,9 @@
 		..()
 
 /obj/structure/reagent_dispensers/cauldron/barrel/attackby(obj/item/W, mob/user, params)
```
Comment (boy2mantwicethefam) [inline code/modules/reagents/reagent_dispenser.dm]: Are you sure you aren't supposed to slap a ..() here

Comment (Irradiation) [reply code/modules/reagents/reagent_dispenser.dm:462]: 🤷‍♂️

Comment (ShiftyRail) [reply code/modules/reagents/reagent_dispenser.dm]: It calls the parent. I'm not sure if it's strictly needed, though. I think generic actions like fingerprints are handled in `preattack`. But it wouldn't change anything to put one here.

**Comment by PrimeDSS13** [comment]
no no no NO NO

**Comment by smalIboss** [comment]
>This incidentally also prevents you from stuffing a gazillion (hostile) mobs into barrels without them being able to escape.

no no no

**Comment by Baronzegit** [comment]
This ain't [content] this is [malice]

STOP THE COUNT

File: code/modules/reagents/reagent_dispenser.dm [hunk 2]
```diff
@@ -437,6 +448,7 @@
 		AM.forceMove(loc)
 
 /obj/structure/reagent_dispensers/cauldron/barrel/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	..()
```
Comment (Kurfursten) [inline code/modules/reagents/reagent_dispenser.dm]: put this at the end so that it comes after special behaviors like grabbing

**Comment by gurfan** [comment]
nein

File: code/modules/reagents/reagent_dispenser.dm [hunk 3]
```diff
@@ -412,17 +412,28 @@
 	desc = "Originally used to store liquids & powder. It is now used as a source of comfort. This one is made of metal."
 	layer = TABLE_LAYER
 	flags = FPRINT | TWOHANDABLE | MUSTTWOHAND // If I end up being coherent enough to make it holdable in-hand
-	throwforce = 40 // Ends up dealing 20~ brute when thrown because thank you, based throw damage formula
```
Comment (Exxion) [inline code/modules/reagents/reagent_dispenser.dm:415]: sneaky

Comment (Kurfursten) [reply code/modules/reagents/reagent_dispenser.dm:415]: most likely just for parity with #28628

**Comment by Irradiation** [comment]
> stuffing a gazillion mobs into barrels is a great feature

It is.

File: code/modules/reagents/reagent_dispenser.dm (hunk 2 above)
Comment (Kurfursten) [reply code/modules/reagents/reagent_dispenser.dm]: I had a closer look at this, you've just added a second attackby for the same object! Don't do that.

Also, you're right, it is handled later, so why are you calling it here at the top??

Comment (Irradiation) [reply code/modules/reagents/reagent_dispenser.dm:451]: Because I had Shifty & Barry telling me to add a ..() at the top 🤷‍♂️

Comment (Kurfursten) [reply code/modules/reagents/reagent_dispenser.dm]: They probably didn't see the other one at the bottom. You doubleplussure need to not have two attackbys on the same object. It works but it's actually the worst thing you can do for code legibility.

File: code/modules/reagents/reagent_dispenser.dm (hunk 3 above)
Comment (Exxion) [reply code/modules/reagents/reagent_dispenser.dm:415]: So leave it to that PR
You don't need parity

Comment (Irradiation) [reply code/modules/reagents/reagent_dispenser.dm:415]: Yes dad 😔 

I shall prepare for unforeseen consequences.

**Comment by gurfan** [comment]
I can't believe this

**Comment by sumireiro-sora** [comment]
It was decided before the PR even went up

**Comment by Irradiation** [comment]
Truth is, Kurf rigged it from the start.

**Comment by PJB3005** [comment]
This change made it so that using barrels for, you knowing, storing and transferring reagents causes you to always attack them *and* have a 1 second cooldown per interaction.

Not great.

**Pull request merged**